dev/test/device_file_io.cu

```diff
@@ -0,0 +1,61 @@
+/*
+Tests device <-> file IO functions
+
+compile and run as (from dev/test directory)
+nvcc -o device_file_io device_file_io.cu && ./device_file_io
+*/
+
+
+#include "../../llmc/cuda_common.h"
+#include <vector>
+#include <random>
+#include <algorithm>
+
+void test(size_t nelem, size_t wt_buf, size_t rd_buf) {
+
+    float* data;
+    cudaCheck(cudaMalloc(&data, nelem*sizeof(float)));
+
+    // generate random array
+    std::vector<float> random_data(nelem);
+    std::mt19937 rng(42);
+    std::uniform_real_distribution<float> dist(-100.f, 100.f);
+    std::generate(random_data.begin(), random_data.end(), [&](){ return dist(rng); });
+
+    cudaCheck(cudaMemcpy(data, random_data.data(), random_data.size()*sizeof(float), cudaMemcpyHostToDevice));
+
+    cudaStream_t stream;
+    cudaStreamCreate(&stream);
+
+    FILE* tmp = fopenCheck("tmp.bin", "w");
```

Comment:
let's remove the file when the test ends in case we want to integrate this into a CI?

not sure what runner machines do with storage (i.e. is it persistent or volatile) - but still a good practice

Reply:
You can assume on the runners it's not persistent.

Reply:
we might still run multiple test cases in one container, so I think cleaning up is still the right choice

Reply:
Yes, good point - was just commenting on the runners. Always cleaning up is a good strategy regardless of persistence. You never know where the code will be used next. Especially, if you run it locally...